src/feature/hs/hs_client.c

```diff
@@ -100,6 +101,45 @@ fetch_status_should_close_socks(hs_client_fetch_status_t status)
   return 1;
 }
 
+/* Return a newly allocated list of all the entry connections that matches the
+ * given service identity pk. If service_identity_pk is NULL, all entry
+ * connections with an hs_ident are returned.
+ *
+ * Caller must free the returned list but does NOT have ownership of the
+ * object inside thus they have to remain untouched. */
+static smartlist_t *
+find_entry_conns(const ed25519_public_key_t *service_identity_pk)
```

Comment:
Suggestion: Let's rename to `find_entry_conns_for_onion_service` or something, to make it a bit more HS-centric? Because initially I thought it just returned everything.

Reply:
Well it is in `hs_client.c` ? ... I mean, it can't really be looking for any other kind of entry connections? Especially when it takes a "service identity key" ?